### PR TITLE
feature: batch_norm_support_5d&basic_contiguous_refactor

### PR DESCRIPTION
**Pull request by yewentao256**
1. batch norm support 5d
2. basic contiguous refactor, support NCHW <-> NHWC, NCDHW <-> NDHWC automatically. However, there are still a lot of things to do towards contiguous, we'll make it in later pr

### CODE REVIEW

File: camb/diopi_helper.hpp [hunk 1]
```diff
@@ -274,6 +290,39 @@ inline DiopiTensor requiresTensor(diopiContextHandle_t ctx, const std::vector<in
...
 
+inline DiopiTensor requiresTensor(diopiContextHandle_t ctx, const std::vector<int64_t>& size, diopiDtype_t dtype, MemoryFormat memory_format) {
+    int64_t dim = size.size();
+    std::vector<int64_t> strides(dim);
+    int64_t stride = 1;
+    if (memory_format == MemoryFormat::Contiguous) {
+        for (size_t i = dim; i > 0; --i) {
+            strides[i - 1] = stride;
+            if (size[i - 1] == 0) continue;
+            if (size[i - 1] == -1) stride = -1;
```
[the remark this answers is not in the document]
Comment (yewentao256) [reply camb/diopi_helper.hpp]: done
